ASIM/dev/ASimTester/filteringTest/filteringTest.py

```diff
@@ -0,0 +1,227 @@
+        # Taking the first value returned in the response
+        selected_value = no_filter_response.tables[0].rows[0][0]
+        value_to_filter = f"\'{selected_value}\'" if param['Type']=="string" else selected_value
+
+        # Performing a filtering by the first value returned in the first response
+        query_with_filter = no_call_query + create_call_with_parameter(param_name, value_to_filter, column_name_in_table)
+        if selected_value=="":
+            query_with_filter = no_call_query + f"query() | where isempty({column_name_in_table}) | summarize count() by {column_name_in_table}\n"
+        filtered_response = self.send_query(query_with_filter)
+        with self.subTest():
+            self.assertNotEqual(0, len(filtered_response.tables[0].rows), f"Parameter: {param_name} - Got no results at all after filtering. Filtered by value: {value_to_filter}")
+        with self.subTest():
+            self.assertEqual(1, len(filtered_response.tables[0].rows), f"Parameter: {param_name} - Expected to have results for only one value after filtering. Filtered by value: {value_to_filter}")
+
+        # Performing a query with a non-existing value, expecting to return no results
+        no_results_query = no_call_query + create_call_with_parameter(param_name, DUMMY_VALUE, column_name_in_table)
+        no_results_response = self.send_query(no_results_query)
+        with self.subTest():
+            self.assertEqual(0, len(no_results_response.tables[0].rows), f"Parameter: {param_name} - Returned results for non existing filter value. Filtered by value: {DUMMY_VALUE}")
+        
+
+        
+    def disabled_test(self, param, no_call_query):
+        disabled_true_query = no_call_query + f"query(disabled=true) | summarize count()\n"
+        disabled_true_response = self.send_query(disabled_true_query)
+        self.assertEqual(0, disabled_true_response.tables[0].rows[0][0], "Expected to return 0 results for disabled=true")
+
+        disabled_false_query = no_call_query + f"query(disabled=false) | summarize count()\n"
+        disabled_false_response = self.send_query(disabled_false_query)
+        self.assertNotEqual(0, disabled_false_response.tables[0].rows[0][0], "Expected to return results for disabled=false")
+
+
+    # Return a set of the columns that will appear in a response of a query call
+    def get_columns_of_parser_answer(self, no_call_query):
+        response = self.send_query(no_call_query + f"query() | getschema\n")
+        columns_set = set()
+        for row in response.tables[0].rows:
+            columns_set.add(row['ColumnName'])
+        return columns_set
+
+
+    def send_query(self, query_str):
+        try:
+            response = client.query_workspace(
+                workspace_id = ws_id,
+                query = query_str,
+                timespan = (start_time, end_time)
+                )
+            if response.status == LogsQueryStatus.PARTIAL:
```

Comment:
>response [](http://example.com/codeflow?start=15&length=8)

Do you also get the failure reason in the response? I think it'll be good to print it for the customer if it's clear enough